tests/end_to_end/Security/test_dual_vm_token_hack.py

```diff
@@ -0,0 +1,166 @@
+import logging
+
+import pytest
+import pytest_asyncio
+
+from kakarot_scripts.utils.kakarot import deploy as deploy_kakarot
+from kakarot_scripts.utils.starknet import deploy as deploy_starknet
+from kakarot_scripts.utils.starknet import get_contract as get_contract_starknet
+from kakarot_scripts.utils.starknet import invoke
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@pytest_asyncio.fixture(scope="package")
+async def starknet_token(owner):
+    address = await deploy_starknet(
+        "StarknetToken",
+        "MyToken",
+        "MTK",
+        18,
+        int(2**256 - 1),
+        owner.starknet_contract.address,
+    )
+    return get_contract_starknet("StarknetToken", address=address)
+
+
+@pytest_asyncio.fixture(
+    scope="package",
+    params=[
+        ("CairoPrecompiles", "DualVmToken"),
+        ("Security", "DualVmTokenWithoutModifier"),
+    ],
+    ids=["Modifier", "NoModifier"],
+)
+async def dual_vm_token(request, kakarot, starknet_token, owner):
```

Comment:
params on fixture makes every function that relies on `dual_vm_token` run twice, one with each version?

Reply:
Yes we want to check that the fix on solidity is working (the nodelegateCallModifier) but also without it that the cairo fix is working.
No sure yet this is the right approach and might need to rethink how to test this.